lib/aruba/announcer.rb

```diff
@@ -0,0 +1,161 @@
+module Aruba
+  # Announcer
+  #
+  # @example Activate your you own channel in cucumber
+  #
+  #   Before('@announce-my-channel') do
+  #     announcer.activate :my_channel
+  #   end
+  #
+  # @example Activate your you own channel in rspec > 3
+  #
+  #   before do
+  #     current_example = context.example
+  #     announcer.activate :my_channel if current_example.metadata[:announce_my_channel]
+  #   end
+  #
+  #   Aruba.announcer.announce(:my_channel, 'my message')
+  #
+  class Announcer
+    # Dev null
+    class NullAnnouncer
+      def announce(*)
+        nil
+      end
+
+      def mode?(*)
+        true
+      end
+    end
+
+    # Announcer using Kernel.puts
+    class KernelPutsAnnouncer
+      def announce(message)
+        Kernel.puts message
+      end
+
+      def mode?(m)
+        :kernel_puts == m
+      end
+    end
+
+    # Announcer using Main#puts
+    class PutsAnnouncer
+      def announce(message)
+        Kernel.puts message
+      end
+
+      def mode?(m)
+        :puts == m
+      end
+    end
+
+    private
+
+    attr_reader :announcers, :default_announcer, :announcer, :channels, :format_strings
+
+    public
+
+    def initialize(_, options)
+      @announcers = []
+      @announcers << PutsAnnouncer.new
+      @announcers << KernelPutsAnnouncer.new
+      @announcers << NullAnnouncer.new
+
+      @default_announcer = @announcers.last
+      @announcer         = @announcers.first
+      @channels          = {}
+      @format_strings    = {}
+
+      @options           = options
+
+      after_init
+    end
+
+    # rubocop:disable Metrics/MethodLength
+    def after_init
+      format_string :directory, '$ cd %s'
+      format_string :command, '$ %s'
+      format_string :environment, '$ export %s=%s"'
+
+      # rubocop:disable Metrics/LineLength
+      if @options[:stdout]
```

Comment:
Those `if`-statements are removed in the upcoming version. I added them to be backwards compatible.